### DISCUSSION TITLE
How to use tokio server-side only for fullstack app?

### DISCUSSION THREAD

**Discussion by dan-da** [DioxusLabs/dioxus#2310] [General]
My webapp needs to invoke RPC calls using the tarpc async RPC framework, which uses tokio.  However it seems that tokio and wasm are not really compatible.

Ok, it seems like that should be fine.  A fullstack app is comprised of a client and server.   So I'd like to use tarpc/tokio from the server (non-wasm) only.

But if i even just add a dep on tarpc/tokio, my fullstack build fails with errors in mio/tokio.

So, how can I build the server (only) of a fullstack app to use tarpc+tokio?     Is it even possible?

**Comment by ealmloff** [comment]
You can add tokio as an optional dependency that is enabled when the server feature is enabled and then use tokio under a `#[cfg(feature = "server")]` block or in a server function

**Reply by dan-da** [reply]
very helpful, thx!
